### DISCUSSION TITLE
is there a way to generate all possible smiles starting just from the chemical formula?

### DISCUSSION THREAD

**Discussion by ccbiozhaw** [rdkit/rdkit#6450] [General]
title says it all. could i input a formula like 
"C10H4N3O8", then get all possible smiles that satisfy this formula?

**Comment by DavidACosgrove** [comment]
It is non-trivial.  Try looking here https://gdb.unibe.ch/downloads/ to see
just how big chemical space is.

On Thu, Jun 8, 2023 at 8:46 AM ccbiozhaw ***@***.***> wrote:

> title says it all. could i input a formula like
> "C10H4N3O8", then get all possible smiles that satisfy this formula?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6450>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FTVB6NSPXL7W3ENERTXKF7OVANCNFSM6AAAAAAY646E7U>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by ccbiozhaw** [reply]
my molecules are larger. however i can already narrow them down to a specific formula. i just need the possible compounds from that

**Comment by DavidACosgrove** [comment]
But if you poke about on that website you will be able to pick up
information about how you might go about it.  I offered that link as a way
into the research area, not the final answer.

On Thu, Jun 8, 2023 at 9:23 AM ccbiozhaw ***@***.***> wrote:

> my molecules are larger. however i can already narrow them down to a
> specific formula. i just need the possible compounds from that
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6450#discussioncomment-6119715>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FXQHOSBZ6NFG55VMYTXKGDZHANCNFSM6AAAAAAY646E7U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
